lib/connection/connection_config.js

```diff
@@ -308,19 +308,18 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
   }
 
   let token = options.token;
-  if (Util.exists(options.token)) {
-    Errors.checkArgumentValid(Util.isString(token),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_OAUTH_TOKEN);
-  }
   if (authenticator === authenticationTypes.OAUTH_AUTHENTICATOR && !Util.string.isNotNullOrEmpty(token)) {
     const tokenFilePath = options.token_file_path ? options.token_file_path : '/snowflake/session/token';
     if (Util.exists(tokenFilePath)) {
       validateOnlyUserReadWritePermission(tokenFilePath);
       token =  fs.readFileSync(tokenFilePath, 'utf-8').trim();
-      Errors.checkArgumentValid(Util.isString(token),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_OAUTH_TOKEN);
+      Logger.getInstance().info('Token used in connection is read from file: %s', tokenFilePath);
     }
   }
+  if (Util.exists(token)) {
```

Comment:
should you change also any test?

Reply:
No. It isn't affect logic. This part is only refactor, the behaviour shouldn't be changed.